src/main/scala/za/co/absa/abris/avro/read/confluent/SchemaManagerFactory.scala

```diff
@@ -17,87 +17,55 @@
 
 package za.co.absa.abris.avro.read.confluent
 
-import java.util
-import io.confluent.common.config.ConfigException
-import io.confluent.kafka.schemaregistry.client.{CachedSchemaRegistryClient, SchemaRegistryClient}
-import io.confluent.kafka.serializers.KafkaAvroDeserializerConfig
 import org.apache.spark.internal.Logging
-import za.co.absa.abris.avro.registry.{AbrisMockSchemaRegistryClient, CustomRegistryClient}
+import za.co.absa.abris.avro.registry.{AbrisRegistryClient, ConfluentRegistryClient}
 import za.co.absa.abris.config.AbrisConfig
+import za.co.absa.commons.annotation.DeveloperApi
 
-import scala.collection.JavaConverters._
 import scala.collection.concurrent
+import scala.util.control.NonFatal
 
 /**
  * This thread-safe factory creates [[SchemaManager]] and also manages the instances of SchemaRegistryClient
  * used by allowing caching of the references in order to avoid creating instances in every call that can be
  * used to cache schemas.
- * This factory also allows us to mock the client for testing purposes.
+ * This factory also allows us to use custom registry client via abris.registryClient.class property.
  */
 object SchemaManagerFactory extends Logging {
 
-  private val clientInstances: concurrent.Map[Map[String,String], SchemaRegistryClient] = concurrent.TrieMap()
+  private val clientInstances: concurrent.Map[Map[String,String], AbrisRegistryClient] = concurrent.TrieMap()
 
-  def addSRClientInstance(configs: Map[String, String], client: SchemaRegistryClient): Unit = {
+  @DeveloperApi
+  def addSRClientInstance(configs: Map[String, String], client: AbrisRegistryClient): Unit = {
     clientInstances.put(configs, client)
   }
 
+  @DeveloperApi
   def resetSRClientInstance(): Unit = {
    clientInstances.clear()
   }
 
   def create(configs: Map[String,String]): SchemaManager = new SchemaManager(getOrCreateRegistryClient(configs))
 
-  private def getOrCreateRegistryClient(configs: Map[String,String]): SchemaRegistryClient = {
+  private def getOrCreateRegistryClient(configs: Map[String,String]): AbrisRegistryClient = {
     clientInstances.getOrElseUpdate(configs, {
-      val settings = new KafkaAvroDeserializerConfig(configs.asJava)
-      val urls = settings.getSchemaRegistryUrls
-      val maxSchemaObject = settings.getMaxSchemasPerSubject
-
```

Comment:
We already have `AbrisConfig.REGISTRY_CLIENT_CLASS` that allows you to theoretically set any registry, even something completely different from confluent. I don't think we need another way of setting mock registry. 

Another issue is that `KafkaAvroDeserializerConfig` is used to get this mock setting which is tied to external code. I would like this dependency to be encapsulated inside the registry client abstraction only.

Reply:
Hmm. I think we should keep that logic somewhere, because exactly this functionality was recently added https://github.com/AbsaOSS/ABRiS/pull/180. If I understand correctly, the dependency on `KafkaAvroDeserializerConfig` is only used to get the schema registry urls. I guess that can also be solved differently, e.g. using `za.co.absa.abris.config.SCHEMA_REGISTRY_URL`

Reply:
Ok for backward compatibility it makes sense. Maybe ConfluentRegistryClient could have factory method that would check this. That way, it would be encapsulated there as well.

Reply:
At the end I did what you suggested, seems to be cleaner approach after all

Reply:
Compared to the original logic, throwing error messages in case of multiple schema registry urls is not covered. The behaviour would be different if someone passed a mock url and a real url. Previously that would have thrown an exception, now it will proceed to create the mock registry. I think that's acceptable. It shouldn't break anything.